Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,251 @@
+            fail_test "PID not matched"
+        fi
+    else
+        fail_test "PID not found in DA_PID"
+    fi
+}
+
+function check_prereqs() {
+    LogMsg "Checking for preconditions"
+
+    if [[ $EUID -ne 0 ]]; then
+        fail_test "This script must be run as root"
+    fi
+
+    verify_distro
+
+    directory_exists=$(verify_directory_exists)
+    if [ "directory_exists" == "true" ]; then
+        SetTestStateAborted
+        exit 0
+    fi
+
+    LogMsg "Preconditions met"
+}
+
+function download_da_linux_installer() {
+    LogMsg "Download Dependency Agent Linux installer"
+
+    if ! curl -L https://aka.ms/dependencyagentlinux -o InstallDependencyAgent-Linux64.bin; then
+        fail_test Download failed
+    fi
+
+    chmod +x InstallDependencyAgent-Linux64.bin
+}
+
+function verify_install_da() {
+    LogMsg "Starting Install tests"
+
+    ./InstallDependencyAgent-Linux64.bin -vme
+
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        fail_test "Install failed with exit code ${ret}"
+    fi
+    LogMsg "Dependency Agent installed successfully"
+
+    verify_expected_file "$uninstaller"
+    verify_expected_file "/etc/init.d/microsoft-dependency-agent"
+    verify_expected_file "/var/opt/microsoft/dependency-agent/log/install.log"
+
+    bin_version=$(./InstallDependencyAgent-Linux64.bin --version | awk '{print $5}')
+    install_log_version=$(sed -n 's/^Dependency Agent version\.revision: //p' /var/opt/microsoft/dependency-agent/log/install.log)
+    if [ "$bin_version" -ne "$install_log_version" ]; then
+        fail_test "Version mismatch between bin version($bin_version) and install log version($install_log_version)"
+    fi
+    LogMsg "Version matches between bin version and install log version"
+
+    LogMsg "Install tests passed successfully"
+}
+
+function verify_uninstall_da() {
+    LogMsg "Starting uninstall tests"
+
+    eval $uninstaller
+
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        fail_test "Uninstall failed with exit code ${ret}"
+    else
+        LogMsg "Uninstalled with exit code 0"
+    fi
+
+    directory_exists=$(verify_directory_exists)
+    if [ "directory_exists" == "true" ]; then
+        fail_test "Directory exists"
+    fi
+
+    LogMsg "Uninstall tests passed successfully"
+}
+
+function generate_network_activity() {
+    LogMsg "Generate network activity"
+    watch -n 5 curl https://microsoft.com &>/dev/null &
+    network_activity_generator_pid=$!
+}
+
+function enable_disable_da(){
+    LogMsg "Starting Enable/Disable DA tests"
+
+    verify_install_da
+
+    # Wait for 30s for DA to start running and check for service.log
+    sleep 30
+    verify_expected_file "/var/opt/microsoft/dependency-agent/log/service.log"
+
+    # Check for service.log.1 file and confirm service.log.2 doesn't exist
+    service_log_time=$(date -r /var/opt/microsoft/dependency-agent/log/service.log +%s)
+    time_limit=$(($service_log_time + 60))
+    if [ "$DISTRO" == "debian"*] || ["$DISTRO" == "ubuntu"* ]; then
```

Comment:
Suggestion: Because we are inspecting the value of `$DISTRO` in the test suite pre-requisites check, consider setting up and distro specific test parameters are the same time.

```
case "$DISTRO" in
    redhat* | centos*)
        readonly enable_timeout=60
        LogMsg "Supported Distro family: $DISTRO"
        ;;
    debian* | ubuntu*)
        readonly enable_timeout=120
        LogMsg "Supported Distro family: $DISTRO"
        ;;
    *)
        fail_test "Unsupported Distro family: $DISTRO"
        ;;
esac
```

Reply:
Adopted the suggestion and refactored the code accordingly.